hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/common/HazelcastSpecialOperator.java

```diff
@@ -31,6 +31,13 @@
  * See {@link HazelcastOperandTypeCheckerAware} for motivation.
  */
 public abstract class HazelcastSpecialOperator extends SqlSpecialOperator implements HazelcastOperandTypeCheckerAware {
+    protected HazelcastSpecialOperator(
```

Comment:
If we use this constructor, we do not wrap the return type inference, which means that we may lose the return type inferred during the validation. All our constructors should have a non-null inference strategy wrapped with `HazelcastReturnTypeInference.wrap`.

Reply:
Constructor removed.